### ISSUE TITLE
[HOLD App#37783] [$250] Multilevel tags - Tag name in selection list shows the other tag name in confirmation page

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#38465]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v.1.4.53-2
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4429997
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
Precondition:
- User is an employee of Collect workspace.
- The Collect workspace has dependant multilevel tag set up
- The multilevel tag is as follow:
First level: State
Second level: Region

1. Go to staging.new.expensify.com
2. Go to workspace chat.
3. Begin manual request flow.
4. In confirmation page, click Show more.
5. Click the first level tag: State.
6. Click the second level tag: Region.

## Expected Result:
The tag name in RHP will show the correct tag name.

## Actual Result:
The tag name in RHP shows the incorrect tag name.
In Step 5, it shows Region when it is State.
In Step 6, it shows State when it is Region.
Different from issue 37783 because this issue occurs in confirmation page only in this build v.1.4.53-2

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/65c874b7-26c9-479d-b140-829a1c881ec3

[Bug6417475_1710758436175!Dependent.-.Multi.Level.tags.NEW.csv](https://github.com/Expensify/App/files/14635711/Bug6417475_1710758436175.Dependent.-.Multi.Level.tags.NEW.csv)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015c4cfe611e62ddef</li>
        <li>Upwork Job ID: 1769707891827982336</li>
        <li>Last Price Increase: 2024-03-18</li>
    </ul>
</details>

**Comment by izarutskaya** [comment]
@puneetlath I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by izarutskaya** [comment]
We think that this bug might be related to the #collect project.

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
 Multilevel tags - Tag name in selection list shows the other tag name in confirmation page

### What is the root cause of that problem?
We order it with orderWeight to get tag list here
https://github.com/Expensify/App/blob/28cfc1409e1f8c310823876553f7b35a75e571e4/src/libs/PolicyUtils.ts#L190-L191
But we do not order before we get the tag name using the unordered list here
https://github.com/Expensify/App/blob/28cfc1409e1f8c310823876553f7b35a75e571e4/src/pages/iou/request/step/IOURequestStepTag.js#L95-L96
### What changes do you think we should make in order to solve the problem?

We should order the policy tag list as in here
https://github.com/Expensify/App/blob/28cfc1409e1f8c310823876553f7b35a75e571e4/src/libs/PolicyUtils.ts#L190-L191

### What alternative solutions did you explore? (Optional)

**Comment by mountiny** [comment]
Thanks! Odd this came as a blocker now. I will demote this also because the multilevel tags are not yet used by customers in NewDot and put on hold for that linked issue

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Looks like the other issue is fixed on staging. Asked for a re-test here: https://expensify.slack.com/archives/C9YU7BX5M/p1714410750527969

**Comment by kbecciv** [comment]
Unable to reproduce it, build 1.4.67.2

https://github.com/Expensify/App/assets/93399543/b9558fc3-55ad-423e-912d-28291c839661

**Comment by puneetlath** [comment]
Great, closing out!
